subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c

```diff
@@ -35,6 +35,16 @@ int mqtt_client_tcp_connect(struct mqtt_client *client)
 		return -errno;
 	}
 
+#if defined(CONFIG_SOCKS)
+	ret = setsockopt(client->transport.tcp.sock,
```

Comment:
This approach will enforce proxy use when `CONFIG_SOCKS` is enabled, making it impossible to make non-proxied connections in this case. We could check lets say `addrlen` here, to check if the user actually set any proxy on this client:
```
if (client->transport.proxy.addrlen != 0) {
        ret = setsockopt(...);
}
```
This would allow the user to use both, proxied or non-proxied connections in the same application.